make install fails with USE_SYSTEM_P7ZIP=1

When compiling 1.12.4 with system p7zip, make install fails with
```
for j in 7z ; do \
        [ -L /build/julia/pkg/julia/usr/lib/julia/$j ] && continue; \
        patchelf --set-rpath '$ORIGIN/.' /build/julia/pkg/julia/usr/lib/julia/$j || exit 1; \
done
patchelf: getting info about '/build/julia/pkg/julia/usr/lib/julia/7z': No such file or directory
make: *** [Makefile:361: install] Error 1
```
Likely caused by https://github.com/JuliaLang/julia/commit/4539b7b4af8c89dadc72bd079965bd1583bb84b4

--- a/base/Makefile
+++ b/base/Makefile
@@ -183,14 +183,14 @@ endif
 endef
 
 # libexec executables
-symlink_p7zip: $(build_bindir)/7z$(EXE)
+symlink_p7zip: $(build_private_libexecdir)/7z$(EXE)
 
 ifneq ($(USE_SYSTEM_P7ZIP),0)
 SYMLINK_SYSTEM_LIBRARIES += symlink_p7zip
 7Z_PATH := $(shell which 7z$(EXE))
 endif
 
-$(build_bindir)/7z$(EXE):
+$(build_private_libexecdir)/7z$(EXE):
 	[ -e "$(7Z_PATH)" ] && \
 	rm -f "$@" && \
 	ln -sf "$(7Z_PATH)" "$@"

